### ISSUE TITLE
Can not restore moving_mean, moving_variance and ExponentialMovingAverage from checkpoint

### ISSUE THREAD

**Issue by fanq15** [tensorpack/tensorpack#880]
When I restore params from a checkpoint, I get the following warnings:
Actually, I can restore the `BatchNorm/beta` and `BatchNorm/gamma`, but I don't know whether it affect the model performance if don't restore the `ExponentialMovingAverage`, `moving_variance`, and `moving_mean`.

And I can change the name https://github.com/tensorpack/tensorpack/blob/fd19f4e21493b8c6dead47aefeaf574879c2ef4b/tensorpack/models/batch_norm.py#L35
to the corresponding name to load the `moving_mean/ExponentialMovingAverage:0`. But I don't know whether it is ok.

```
[0831 07:13:22 @sessinit.py:90] WRN The following variables are in the checkpoint, but not found in the graph: 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/beta/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/gamma/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/moving_mean/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/moving_mean:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/moving_variance/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/moving_variance:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/weights/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/beta/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/gamma/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/moving_mean/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/moving_mean:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/moving_variance/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/moving_variance:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/weights/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/BatchNorm/beta/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/BatchNorm/gamma/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/BatchNorm/moving_mean/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/BatchNorm/moving_mean:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/BatchNorm/moving_variance/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/BatchNorm/moving_variance:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv3/weights/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/BatchNorm/beta/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/BatchNorm/gamma/ExponentialMovingAverage:0,
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/BatchNorm/moving_mean/ExponentialMovingAverage:0,
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/BatchNorm/moving_mean:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/BatchNorm/moving_variance/ExponentialMovingAverage:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/BatchNorm/moving_variance:0, 
resnet_v1_101/block1/unit_1/bottleneck_v1/shortcut/weights/ExponentialMovingAverage:0, 

```

```
[0831 06:36:41 @sessinit.py:90] WRN **The following variables are in the graph, but not found in the checkpoint:**

resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_1/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_2/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_2/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_2/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_2/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_3/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_3/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_3/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block1/unit_3/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_1/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_1/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_1/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_1/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_2/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_2/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_2/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_2/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_3/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_3/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_3/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_3/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_4/bottleneck_v1/conv1/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_4/bottleneck_v1/conv1/BatchNorm/variance/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_4/bottleneck_v1/conv2/BatchNorm/mean/ExponentialMovingAverage, 
resnet_v1_101/block2/unit_4/bottleneck_v1/conv2/BatchNorm/variance/ExponentialMovingAverage, 

```

**Comment by ppwwyyxx** [comment]
Please post issues following the issue template (New Issue -> Unexpected Problems).

It looks like you're loading a model that does not match your graph definition, and from the logs it looks like the graph definition is not using any tensorpack layers. So the issue does not seem to be related to tensorpack.

**Comment by fanq15** [comment]
Sorry about my unsuitable issue...
I define my graph using tensorpack layer and load a tensorflow model. I change some tensor names to load it correctly, such as: change `W` to `weights` and `b` to `biases`. Actually, I don't know why the tensor names in tensorpack's Conv2D is not corresponding to the tensorflow's. Even I go through the entire tensorpack code, I can not find anything about `ExponentialMovingAverage`, and there are nothing in tensorpack corresponding to it, so I cannot adapt it through change the code.

**Comment by ppwwyyxx** [comment]
Since your model is not originally defined by tensorpack's layer, you should not use tensorpack's layer to define it when you load it. You should refer to whoever provides the model for the code that defines the model.
